### PR TITLE
add SHA to PR/master version, add build_type (stable/master/pr/dev)

### PR DESCRIPTION
**Pull request by birarda**

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/9680/app_13148_PR_45de15656055dfba06666fe2390788ad015e2836.apk

File: cmake/macros/SetPackagingParameters.cmake [hunk 1]
```diff
@@ -73,6 +70,27 @@ macro(SET_PACKAGING_PARAMETERS)
     add_definitions(-DDEV_BUILD)
   endif ()
 
+  if (PRODUCTION_BUILD OR PR_BUILD AND NOT STABLE_BUILD)
```
Comment (Atlante45) [inline cmake/macros/SetPackagingParameters.cmake]: Don't you need to put parenthesis around that?

File: cmake/macros/SetPackagingParameters.cmake [hunk 2]
```diff
@@ -73,6 +70,27 @@ macro(SET_PACKAGING_PARAMETERS)
...
+      OUTPUT_STRIP_TRAILING_WHITESPACE
+    )
+
+    set(BUILD_VERSION "${BUILD_VERSION}-${GIT_COMMIT_HASH}")
+  endif ()
+
+  # if STABLE_BUILD is 1, PRODUCTION_BUILD must be 1 and
+  # DEV_BUILD and PR_BUILD must be 0
+  if (STABLE_BUILD)
+    if (NOT PRODUCTION_BUILD OR PR_BUILD OR DEV_BUILD)
```
Comment (Atlante45) [inline cmake/macros/SetPackagingParameters.cmake]: Same here.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/9684/app_13148_PR_6599441f6050a1bfacd668fb50f0afd61f928f22.apk

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/9689/app_13148_PR_da89e29e43bcd17188a64c7f05fd7bbff7d56b18.apk

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/9731/HighFidelity-Beta-PR13148-a391ec12e10d12d3145467602b902fa3561629a1.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9731/HighFidelity-Beta-PR13148-${GITHUB_COMMIT}.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9731/HighFidelity-Beta-Interface-PR13148-${GITHUB_COMMIT}.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9731/HighFidelity-Beta-PR13148-a391ec12e10d12d3145467602b902fa3561629a1.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9731/HighFidelity-Beta-Interface-PR13148-a391ec12e10d12d3145467602b902fa3561629a1.exe

**Comment by marko8904** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/9739/HighFidelity-Beta-PR13148-d57d50c2c13cbea96d73462c5052dbc35373e8bf.apk

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/10088/HighFidelity-Beta-PR13148-2c9648daaf2ab2692203fff170637d694c5220ac.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10088/HighFidelity-Beta-PR13148-2c9648daaf2ab2692203fff170637d694c5220ac.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10088/HighFidelity-Beta-Interface-PR13148-2c9648daaf2ab2692203fff170637d694c5220ac.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10088/HighFidelity-Beta-PR13148-2c9648daaf2ab2692203fff170637d694c5220ac.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10088/HighFidelity-Beta-Interface-PR13148-2c9648daaf2ab2692203fff170637d694c5220ac.exe

**Comment by marko8904** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/10213/HighFidelity-Beta-PR13148-2c9648.apk

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/10214/HighFidelity-Beta-PR13148-2c9648d.apk

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/10226/HighFidelity-Beta-PR13148-33a122c.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10227/HighFidelity-Beta-PR13148-b9e9c95.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10227/HighFidelity-Beta-Interface-PR13148-b9e9c95.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10227/HighFidelity-Beta-PR13148-b9e9c95.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10227/HighFidelity-Beta-Interface-PR13148-b9e9c95.exe

**Comment by birarda** [comment]
#### Test Plan

In the steps below $SHA is replaced by the abbreviated SHA of the installed PR build.

##### PR Builds

1. Install this PR. Perform a custom installation. Install shortcuts for both the Sandbox and Interface. Confirm that during the install the version appears as "High Fidelity - PR13148" and does not appear as "High Fidelity -PR13148-$SHA".
2. Launch Interface from the PR. Confirm that the application title bar is suffixed by " - "Build PR13148-$SHA".
3. Run the Sandbox from the PR from a protocol incompatible version (ex: current stable). Confirm that you are unable to connect because of a protocol version mismatch and the reported version of the server is "Domain version: PR13148-$SHA".
4. Confirm that the following folders exist: `%APPDATA%\Roaming\High Fidelity - PR13148`, `%APPDATA%\Local\High Fidelity - PR13148`.
5. Confirm that the following folders do not exist: `%APPDATA%\Roaming\High Fidelity - PR13148-$SHA`, `%APPDATA%\Local\High Fidelity - PR13148-$SHA`.
6. Confirm that the start menu and desktop shortcuts are suffixed by "- PR13148" and not "- PR13148-$SHA".
7. Confirm that the uninstall entry in "Apps & Features" in Windows is for "High Fidelity - PR13148" and not "High Fidelity - PR13148-$SHA".

##### Master Builds

http://hifi-public.s3.amazonaws.com/birarda/HighFidelity-Beta-13148-b9e9c952de.exe

1. Install the faked master/nightly build linked above. Perform a custom installation. Install shortcuts for both the Sandbox and Interface. Confirm that during the install the version appears as "High Fidelity" and does not appear as "High Fidelity - 13148-$SHA".
2. Launch Interface. Confirm that the application title bar is suffixed by " - "Build 13148-$SHA".
3. Run the Sandbox and connect from a protocol incompatible version (ex: current stable). Confirm that you are unable to connect because of a protocol version mismatch and the reported version of the server is "Domain version: 13148-$SHA".
4. Confirm that the following folders exist: `%APPDATA%\Roaming\High Fidelity`, `%APPDATA%\Local\High Fidelity`.
5. Confirm that the following folders do not exist: `%APPDATA%\Roaming\High Fidelity - 13148-$SHA`, `%APPDATA%\Local\High Fidelity - 13148-$SHA`.
6. Confirm that the start menu and desktop shortcuts are not suffixed by " - 13148-$SHA".

##### Stable Builds

http://hifi-public.s3.amazonaws.com/birarda/HighFidelity-Beta-0.14.0.exe

1. Install the faked stable build linked above. Perform a custom installation. Install shortcuts for both the Sandbox and Interface. Confirm that during the install the version appears as "High Fidelity" and does not appear as "High Fidelity - 0.14.0-$SHA".
2. Launch Interface. Ignore that you are prompted to update. Confirm that the application title bar is suffixed by " - "Version 0.14.0". 
3. Run the Sandbox and connect from a protocol incompatible version (ex: current stable). Confirm that you are unable to connect because of a protocol version mismatch and the reported version of the server is "Domain version: 0.14.0".
4. Confirm that the following folders exist: `%APPDATA%\Roaming\High Fidelity`, `%APPDATA%\Local\High Fidelity`.
5. Confirm that the following folders do not exist: `%APPDATA%\Roaming\High Fidelity - 0.14.0-$SHA`, `%APPDATA%\Local\High Fidelity - 0.14.0-$SHA`.
6. Confirm that the start menu and desktop shortcuts are not suffixed by " - 0.14.0-$SHA".

**Comment by qligmaloney** [comment]
Currently testing

**Comment by qligmaloney** [comment]
PASS @conklin94122 SHA displays on Interface window and on protocol mismatch but not on App Data files or when Installing

**Pull request merged**